### PR TITLE
Bugfix/caching of paymentmethodsettings issue

### PR DESCRIPTION
**Pull request by ferryvandegraaf**
# Describe your changes

Paymentmethod settings is always cached, something you do not always want. In case there are very much orders (multiple in a second) we don't want to cache this model because the succesurl contains the orderId and this orderId must be different for each payment.

Second bugfix is that Query caching for the GetPaymentMethodAsync function never worked because the parameters were not cleared. So each time the query was different which should not be the case because the only parameter in this query is the id of the payment method.

## Type of change

Please check only ONE option.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested?

I tested with query caching on and loaded the same functionality twice, then it was fixed, second time the caching worked fine.

# Checklist before requesting a review
- [x] I have reviewed and tested my changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I selected `develop` as the base branch and not `main`, or the pull request is a hotfix that needs to be done directly on `main`
- [x] I double checked all my changes and they contain no temporary test code, no code that is commented out and no changes that are not part of this branch
- [ ] I added new unit tests for my changes if applicable

# Related pull requests

Add any open pull requests from other projects that are related to this pull request that should be merged along with this one. If there are none, you can simply say "none" or "N/A", or just leave this section empty.

# Link to Asana ticket

Add a link to the Asana ticket. Note that **_ONLY_** the URL should be added, not the name of the ticket!

### CODE REVIEW

**Comment by HansHappyHorizon** [comment]
Please fill out the entire form instead of only checking some boxes (summary of changes and explanation of how the changes were tested).

**Comment by ferryvandegraaf** [comment]
Description added.

File: GeeksCoreLibrary/Components/OrderProcess/Services/CachedOrderProcessesService.cs [hunk 1]
```diff
@@ -73,6 +73,9 @@ public override async Task<List<PaymentMethodSettingsModel>> GetPaymentMethodsAs
         public override async Task<PaymentMethodSettingsModel> GetPaymentMethodAsync(ulong paymentMethodId)
         {
             var cacheName = $"OrderProcessGetPaymentMethodAsync_{paymentMethodId}_{branchesService.GetDatabaseNameFromCookie()}";
+            if (gclSettings.DefaultOrderProcessCacheDuration.TotalSeconds==0)
+                return await orderProcessesService.GetPaymentMethodAsync(paymentMethodId);
```
Comment (GilianJuice) [inline GeeksCoreLibrary/Components/OrderProcess/Services/CachedOrderProcessesService.cs:77]: Please add brackets and spaces. It should be like this:
```C#
if (gclSettings.DefaultOrderProcessCacheDuration.TotalSeconds == 0)
{
    return await orderProcessesService.GetPaymentMethodAsync(paymentMethodId);
}
```

**Review by GilianJuice — CHANGES_REQUESTED** [review]
Firstly, I don't see any changes for the first part of your description, about not caching payment settings anymore? Did you forget something or do I understand your description wrongly?

Secondly, this PR contains changes from other PRs that have nothing to do with what you described. They even contain changes from a different person. Please remove all changes that are not related.

**Comment by GilianJuice** [comment]
Closing this PR because it's outdated. If we ever want these changes, we'll copy them manually from In-vista's fork.

**Pull request closed without merging**